### DISCUSSION TITLE
Overriding the default HTML renderer logic

### DISCUSSION THREAD

**Discussion by Bios-Marcel** [yuin/goldmark#374] [General]
Hey,

I am unsure whether I am missunderstanding something here, but this doesn't quite do what I would expect:

```go
type customHtmlRenderer struct {
	*html.Renderer
}

// RegisterFuncs implements NodeRenderer.RegisterFuncs .
func (r *customHtmlRenderer) RegisterFuncs(reg renderer.NodeRendererFuncRegisterer) {
	r.Renderer.RegisterFuncs(reg)
	reg.Register(ast.KindFencedCodeBlock, r.renderFencedCodeBlock)
	reg.Register(ast.KindHeading, r.renderHeading)
}
```

For fenced code blocks is correctly calls my function, but for headings it doesn't.
No panic, no output, nothing.

Headings are those, right?

```md
# A

## B

### C
```

**Comment by abhinav** [comment]
@Bios-Marcel What priority are you registering your renderer with? The default HTML renderer uses a priority of 1000, so you need to use something lower than that.

**Reply by Bios-Marcel** [reply]
No matter whether I set the priority low or high, it doesn't get called. I set it to a value of under 1000, which is claimed to be the default value.

It gets called for some types, such as fenced code blocks, but never for headings. Seemed like a bug, but I thought something that obvious can't be a bug, so I am probably missunderstanding something.

**EDIT** Anyway ... for some reason I am doing something differently now and god knows what :D
It works. No matter what priority I set. I'll close this. Thanks anyway.

**Reply by abhinav** [reply]
FWIW, the following calls renderHeading correctly.

```go
package main

import (
	"fmt"
	"log"
	"os"

	"github.com/yuin/goldmark"
	"github.com/yuin/goldmark/ast"
	"github.com/yuin/goldmark/renderer"
	"github.com/yuin/goldmark/util"
)

func main() {
	md := goldmark.New(
		goldmark.WithRendererOptions(
			renderer.WithNodeRenderers(
				util.Prioritized(&myHeadingRenderer{}, 500),
			),
		),
	)

	if err := md.Convert([]byte("# A\n## B\n### C\n"), os.Stdout); err != nil {
		log.Fatal(err)
	}
}

type myHeadingRenderer struct{}

var _ renderer.NodeRenderer = &myHeadingRenderer{}

func (r *myHeadingRenderer) RegisterFuncs(reg renderer.NodeRendererFuncRegisterer) {
	reg.Register(ast.KindHeading, r.renderHeading)
}

func (r *myHeadingRenderer) renderHeading(w util.BufWriter, source []byte, node ast.Node, entering bool) (ast.WalkStatus, error) {
	if entering {
		fmt.Fprintf(w, "HEADING %d: ", node.(*ast.Heading).Level)
	} else {
		w.WriteString("\n")
	}
	return ast.WalkContinue, nil
}
```

It also does so if I register a custom html.Renderer first.

**Reply by Bios-Marcel** [reply]
Yeah, I really don't know what I did back there :D
